From 16cdcc469bd2c69139495b00af05b97009c94f1b Mon Sep 17 00:00:00 2001
From: xiaoming <2014500726@smail.xtu.edu.cn>
Date: Tue, 31 Aug 2021 14:58:59 +0800
Subject: [PATCH] add quard_star patch

---
 Makefile | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/Makefile b/Makefile
index 6920d7f..0afdf62 100644
--- a/Makefile
+++ b/Makefile
@@ -29,6 +29,8 @@ CSCOPE	:= cscope
 TAGS	:= ctags
 INSTALL := install
 
+LIBFDT_PATH := /usr
+
 prefix = $(HOME)
 bindir_relative = bin
 bindir = $(prefix)/$(bindir_relative)
@@ -341,13 +343,13 @@ $(warning No static libc found. Skipping guest init)
 endif
 
 ifeq (y,$(ARCH_WANT_LIBFDT))
-	ifneq ($(call try-build,$(SOURCE_LIBFDT),$(CFLAGS),-lfdt),y)
+	ifneq ($(call try-build,$(SOURCE_LIBFDT),$(CFLAGS) -I$(LIBFDT_PATH)/include,-L$(LIBFDT_PATH)/lib -lfdt),y)
           $(error No libfdt found. Please install libfdt-dev package)
 	else
-		CFLAGS_DYNOPT	+= -DCONFIG_HAS_LIBFDT
-		CFLAGS_STATOPT	+= -DCONFIG_HAS_LIBFDT
-		LIBS_DYNOPT	+= -lfdt
-		LIBS_STATOPT	+= -lfdt
+		CFLAGS_DYNOPT	+= -I$(LIBFDT_PATH)/include -DCONFIG_HAS_LIBFDT
+		CFLAGS_STATOPT	+= -I$(LIBFDT_PATH)/include -DCONFIG_HAS_LIBFDT
+		LIBS_DYNOPT	    += -L$(LIBFDT_PATH)/lib -lfdt
+		LIBS_STATOPT	+= -L$(LIBFDT_PATH)/lib -lfdt
 	endif
 endif
 
-- 
2.17.1

