From 09d014421e4cfc1e7a6184b6899c1682562addd1 Mon Sep 17 00:00:00 2001
From: xiaoming <2014500726@smail.xtu.edu.cn>
Date: Tue, 31 Aug 2021 21:17:44 +0800
Subject: [PATCH] add quard_star patch

---
 Makefile     | 12 +++++++-----
 guest/init.c |  2 +-
 2 files changed, 8 insertions(+), 6 deletions(-)

diff --git a/Makefile b/Makefile
index 6920d7f..305440e 100644
--- a/Makefile
+++ b/Makefile
@@ -29,6 +29,8 @@ CSCOPE	:= cscope
 TAGS	:= ctags
 INSTALL := install
 
+LIBFDT_PATH := /usr
+
 prefix = $(HOME)
 bindir_relative = bin
 bindir = $(prefix)/$(bindir_relative)
@@ -341,13 +343,13 @@ $(warning No static libc found. Skipping guest init)
 endif
 
 ifeq (y,$(ARCH_WANT_LIBFDT))
-	ifneq ($(call try-build,$(SOURCE_LIBFDT),$(CFLAGS),-lfdt),y)
+	ifneq ($(call try-build,$(SOURCE_LIBFDT),$(CFLAGS) -I$(LIBFDT_PATH)/include,-L$(LIBFDT_PATH)/lib -lfdt),y)
           $(error No libfdt found. Please install libfdt-dev package)
 	else
-		CFLAGS_DYNOPT	+= -DCONFIG_HAS_LIBFDT
-		CFLAGS_STATOPT	+= -DCONFIG_HAS_LIBFDT
-		LIBS_DYNOPT	+= -lfdt
-		LIBS_STATOPT	+= -lfdt
+		CFLAGS_DYNOPT	+= -I$(LIBFDT_PATH)/include -DCONFIG_HAS_LIBFDT
+		CFLAGS_STATOPT	+= -I$(LIBFDT_PATH)/include -DCONFIG_HAS_LIBFDT
+		LIBS_DYNOPT	    += -L$(LIBFDT_PATH)/lib -lfdt
+		LIBS_STATOPT	+= -L$(LIBFDT_PATH)/lib -lfdt
 	endif
 endif
 
diff --git a/guest/init.c b/guest/init.c
index 52f6567..e69e42f 100644
--- a/guest/init.c
+++ b/guest/init.c
@@ -73,7 +73,7 @@ int main(int argc, char *argv[])
 	}
 
 	sync();
-	reboot(RB_AUTOBOOT);
+	reboot(RB_HALT_SYSTEM);
 
 	printf("Init failed: %s\n", strerror(errno));
 
-- 
2.17.1

