--- ./config.sub	2022-03-28 12:59:43.440841399 +0800
+++ ./config.sub	2022-03-28 12:59:51.108622228 +0800
@@ -301,6 +301,7 @@
 	| pdp10 | pdp11 | pj | pjl \
 	| powerpc | powerpc64 | powerpc64le | powerpcle \
 	| pyramid \
+	| riscv32 | riscv64 \
 	| rl78 | rx \
 	| score \
 	| sh | sh[1234] | sh[24]a | sh[24]aeb | sh[23]e | sh[34]eb | sheb | shbe | shle | sh[1234]le | sh3ele \
@@ -419,6 +420,7 @@
 	| pdp10-* | pdp11-* | pj-* | pjl-* | pn-* | power-* \
 	| powerpc-* | powerpc64-* | powerpc64le-* | powerpcle-* \
 	| pyramid-* \
+	| riscv32-* | riscv64-* \
 	| rl78-* | romp-* | rs6000-* | rx-* \
 	| sh-* | sh[1234]-* | sh[24]a-* | sh[24]aeb-* | sh[23]e-* | sh[34]eb-* | sheb-* | shbe-* \
 	| shle-* | sh[1234]le-* | sh3ele-* | sh64-* | sh64le-* \
--- ./configure	2022-06-21 17:46:18.280324059 +0800
+++ ./configure	2022-06-21 17:46:27.292678210 +0800
@@ -6151,16 +6151,6 @@
 	fi
 fi
 
-if test "$libdir" = "\${exec_prefix}/lib"; then
-	case "$host_os_name" in
-		linux*)
-			if test -d /usr/lib64 -a ! -d /usr/lib64/fakeroot; then
-				libdir="$exec_prefix/lib64"
-			fi
-			;;
-	esac
-fi
-
 
 # Check whether --with-privateinclude was given.
 if test "${with_privateinclude+set}" = set; then :
